src/main/java/mate/academy/bookshop/service/impl/BookServiceImpl.java

```diff
@@ -23,13 +24,44 @@ public BookDto save(CreateBookRequestDto createBookRequestDto) {
         return bookMapper.toDto(bookRepository.save(book));
     }
 
+    @Override
+    public BookDto update(Long id, UpdateBookRequestDto updateBookRequestDto) {
+        Book createdBook = bookRepository.findById(id).orElseThrow(()
+                -> new EntityNotFoundException("Can't find entity by id: " + id));
+        if (updateBookRequestDto.getTitle() != null) {
+            createdBook.setTitle(updateBookRequestDto.getTitle());
+        }
+        if (updateBookRequestDto.getAuthor() != null) {
+            createdBook.setAuthor(updateBookRequestDto.getAuthor());
+        }
+        if (updateBookRequestDto.getIsbn() != null) {
+            createdBook.setIsbn(updateBookRequestDto.getIsbn());
+        }
+        if (updateBookRequestDto.getPrice() != null) {
+            createdBook.setPrice(updateBookRequestDto.getPrice());
+        }
+        if (updateBookRequestDto.getDescription() != null) {
+            createdBook.setDescription(updateBookRequestDto.getDescription());
+        }
+        if (updateBookRequestDto.getCoverImage() != null) {
+            createdBook.setCoverImage(updateBookRequestDto.getCoverImage());
+        }
+        return bookMapper.toDto(bookRepository.save(createdBook));

```

Comment:
it can be a bit simpler than that:
```suggestion
        var updatedBook = bookMapper.toModel(requestDto);
        updatedBook.setId(id);
        return bookMapper.toDto(repo.save(updatedBook));
```
because hibernate will know that we need to update the fields of the existing book by the fact that we have set an id of an already existing book, and we also do not have to check for null every single field, just save what we have received because it is PUT, not PATCH